public/Aryan Suri/Sidebar/sidebar.js

```diff
@@ -640,265 +556,228 @@ async function Sidebar() {
+        var propIndex;
+        var sourceIndex;
+        var sourceFactor;
+        var targetIndex;
+        var targetFactor;
+        var result;
+        // Start by checking which this.property we are working in...
+        propIndex = document.property_form.the_menu.selectedIndex;
+        // Let's determine what this.unit are we converting FROM (i.e. source) and the this.factor needed to convert that this.unit to the base this.unit.
+        sourceIndex = sourceForm.unit_menu.selectedIndex;
+        sourceFactor = this.factor[propIndex][sourceIndex];
+        // Cool! Let's do the same thing for the target this.unit - the units we are converting TO:
+        targetIndex = targetForm.unit_menu.selectedIndex;
+        targetFactor = this.factor[propIndex][targetIndex];
+        // Simple, huh? let's do the math: a) convert the source TO the base this.unit: (The input has been checked by the CalculateUnit function).
+        result = sourceForm.unit_input.value;
+        // Handle Temperature increments!
+        if (this.property[propIndex] == "Temperature") {
+            result = parseFloat(result) + this.tempIncrement[sourceIndex];
+        }
+        result = result * sourceFactor;
+        // not done yet... now, b) use the targetFactor to convert FROM the base this.unit
+        // to the target this.unit...
+        result = result / targetFactor;
+        // Again, handle Temperature increments!
+        if (this.property[propIndex] == "Temperature") {
+            result = parseFloat(result) - this.tempIncrement[targetIndex];
+        }
+        // Ta-da! All that's left is to update the target input box:
+        targetForm.unit_input.value = result;
+    }
 }
-
 function rtdefault() {
-	$('section.mt-content-container p').css("font-size", 1.1 + "rem");
-	$("#size").val("1.1");
-	$("#slider-page-width").val("0");
-	$("#toggler-text").attr("class", "toggler");
-	$('section.mt-content-container').css("margin-left", 0 + "px");
-	$('section.mt-content-container').css("margin-right", 0 + "px");
-	sessionStorage.setItem('page_width', 0);
-	$('section.mt-content-container p').css("text-align", "justify");
-};
-
-
-
-
+    $('section.mt-content-container p').css("font-size", 1.1 + "rem");
+    $("#size").val("1.1");
+    $("#slider-page-width").val("0");
+    $("#toggler-text").attr("class", "toggler");
+    $('section.mt-content-container').css("margin-left", 0 + "px");
+    $('section.mt-content-container').css("margin-right", 0 + "px");
+    sessionStorage.setItem('page_width', '0');
+    $('section.mt-content-container p').css("text-align", "justify");
+}
+;
+function saveBookmark() {
+    const TITLE = document.getElementById("titleHolder").innerText;
+    const URL = window.location.href;
+    if (URL) {
+        sessionStorage.setItem("Title", TITLE);
+        sessionStorage.setItem("Bookmark", URL);
+        createBookmarks();
+    }
+    else {
+        console.log("Bookmark created already!");
+    }
```

Comment:
This check does not appear to be working and will always return true.

Reply:
Will fix this